src/libstd/sys_common/net.rs

```diff
@@ -53,6 +53,8 @@ use libc::MSG_NOSIGNAL;
               target_os = "haiku", target_os = "bitrig")))]
 const MSG_NOSIGNAL: c_int = 0x0;
 
+const MSG_PEEK: c_int = 0x2;
```

Comment:
Yes, let's use libc if possible. I've added the flag only very recently in https://github.com/rust-lang/libc/pull/492. How is best to make this transition? Should we hold off on this PR until the next libc release?